### PR TITLE
Convert some run_command() string args to lists

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
This is a general cleanup refactoring.

##### ISSUE TYPE
- Refactoring Pull Request

##### COMPONENT NAME
various

### CODE REVIEW

File: plugins/modules/parted.py [hunk 1]
```diff
@@ -580,6 +580,7 @@ def parted(script, device, align):
         script_option = '-s'
 
     if script and not module.check_mode:
+        # TODO: convert run_comand() argument to list!
```
Comment (felixfontein) [inline plugins/modules/parted.py:583]: This one has some history (#3823) and definitely won't happen in this PR.

**Comment by felixfontein** [comment]
This is finally ready for review! :tada:

**Review by russoz — APPROVED** [review]
Other than a minor readability nit, LGTM

File: plugins/modules/pkgin.py [hunk 2]
```diff
@@ -145,18 +145,18 @@ def query_package(module, name):
...
         splitchar = ';'
     else:
-        pflag = ''
+        pflag = []
         splitchar = ' '
 
     # Use "pkgin search" to find the package. The regular expression will
     # only match on the complete name.
-    rc, out, err = module.run_command("%s %s search \"^%s$\"" % (PKGIN_PATH, pflag, name))
+    rc, out, err = module.run_command([PKGIN_PATH] + pflag + ["search", "^%s$" % (name, )])
```
[the remark this answers is not in the document]
Comment (felixfontein) [reply plugins/modules/pkgin.py:159]: It's easier to read, but has a different semantic :) If `name` happens to be a tuple, only its first element will be used.

(That's one of the reasons why I prefer `'...'.format(...)` over `'...' % ...`.)

[the remark this answers is not in the document]
Comment (russoz) [reply plugins/modules/pkgin.py]: ... and if it formats a tuple, or worse an array, in the command line, that might lead us into unexpected behaviors in bash. And, more to the point, do we expect those values to ever be tuples?

**Comment by russoz** [comment]
LGTM

File: plugins/modules/pkgin.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (felixfontein) [reply plugins/modules/pkgin.py]: True. Unless the receiving program understands Python lists/tuples ;) But I checked all locations before adidng that commit, the values passed should always be strings.

**Comment by felixfontein** [comment]
@russoz thanks for reviewing this!

**Pull request merged**